assayist/processor/loose_rpm_analyzer.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import glob
+import os
+
+from assayist.common.models import content
+from assayist.processor.base import Analyzer
+from assayist.processor.logging import log
+
+
+class LooseRpmAnalyzer(Analyzer):
+    """Analyze RPMs embedded in the build artifacts."""
+
+    def run(self):
+        """Start the loose RPM analyzer."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+
+        # Remember current directory
+        orig_dir = os.getcwd()
+
+        # Switch to dir of all unpacked content to search for RPM files
+        unpacked_content_path = os.path.join(self.input_dir, self.UNPACKED_ARCHIVES_DIR)
+        os.chdir(unpacked_content_path)
```

Comment:
Why are you changing directory instead of doing `for archive_type in os.listdir(unpacked_content_path):`?

Reply:
Because then I have to carry over the `unpacked_content_path` through the other two for loops as well. But I guess it looks better than mucking around with `chdir`.